### PR TITLE
Max pa fix

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
This is needed for platforms that use dynamic shared memory and rely on `register_ddr()` or FDT to describe that memory. This currently seems to be the `plat-vexpress` and `plat-stm` platforms.
I've noticed the problem myself on both Juno and FVP.

Prior to this it has been reported in https://github.com/OP-TEE/optee_os/issues/2402 together with a good analysis by @fractalclone

File: core/arch/arm/mm/core_mmu_lpae.c [hunk 1]
```diff
@@ -468,11 +438,40 @@ void core_init_mmu_regs(void)
 #endif /*ARM32*/
 
 #ifdef ARM64
+static unsigned int calc_physical_addr_size_bits(uint64_t max_addr)
+{
+	/* Physical address can't exceed 48 bits */
+	assert(!(max_addr & ADDR_MASK_48_TO_63));
```
Comment (jforissier) [inline core/arch/arm/mm/core_mmu_lpae.c]: `panic()`?

Comment (jenswi-linaro) [reply core/arch/arm/mm/core_mmu_lpae.c]: Makes sense, I'll fix

**Comment by jforissier** [comment]
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
Squashed and tag applied

**Pull request merged**